crates/matrix-sdk-indexeddb/src/state_store/mod.rs

```diff
@@ -741,37 +815,87 @@ impl_state_store!({
         if !changes.redactions.is_empty() {
             let state = tx.object_store(keys::ROOM_STATE)?;
             let room_info = tx.object_store(keys::ROOM_INFOS)?;
+            let raw_room_infos = tx.object_store(keys::RAW_ROOM_INFOS)?;
 
             for (room_id, redactions) in &changes.redactions {
+                // TODO: Load room version lazily, only if it is necessary
```

Comment:
I removed the lazy loading to make it simpler while implementing, but I should add it back in. The approach from the sqlite make_room_version closure doesn't work here because it's async

Reply:
With the last set of changes, I had to make the sqlite version non-async too, to allow passing the room version into the base_info.handle_redaction function.